python/index_writer.py

```diff
@@ -0,0 +1,55 @@
+import index_writer_py
+from mpi4py import MPI
+import logging
+import sys
+import traceback
+
+def get_logger():
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    ch = logging.StreamHandler(sys.stdout)
+    ch.setFormatter(formatter)
+    logger.addHandler(ch)
+    return logger
+
+logger = get_logger()
+
+def init_mpi():
+    logger.info("Initializing MPI")
+    try:
+        index_writer_py.init_mpi()
+        logger.info("MPI initialized")
+    except Exception as e:
+        logger.error(f"Error initializing MPI: {e}")
+        logger.error(traceback.format_exc())
+        raise
```

Comment:
This seems unnecessary. Can we get away without that?

BTW, logging in Python has an [`exception(` function that should add tracebacks, too](https://docs.python.org/3/library/logging.html#logging.Logger.exception).